frontend/z3_types.py

```diff
@@ -79,8 +78,7 @@ def __init__(self, config):
             ForAll([x, y], Implies(And(self.subtype(x, y), self.subtype(y, x)), x == y)),
             ForAll([x, y, z],
                    Implies(And(self.extends(x, z), self.extends(y, z), x != y),
-                           And(self.not_subtype(x, y), self.not_subtype(y, x)))),
-            ForAll([x, y, z], Implies(And(self.subtype(x, y), self.not_subtype(y, z)), Not(self.subtype(x, z)))),
```

Comment:
After removing this implication, some logical errors appeared in the subtyping relationships. For example, `string` can now be a subtype of `int`.
Accordingly, the following program gives `sat` whereas it should be `unsat`

```python
x = "string"
y = [1,2,3]
y[x]
```